src/my_project/tagging/validator.py

```
@@ -182,43 +172,14 @@ def __init__(
-        logger.info("Starting year checks: ")
-
-        def check_year(item: ComicVineResponse) -> bool:
-            year = int(item.date_added[:4])
-            return abs(year - self.expected_info.pub_year) <= 4
-
-        return self.filter_results(check_year)
+    @staticmethod
+    def get_year(data: ComicVineIssueStruct) -> int:
+        year = (data.cover_date or data.store_date or "")[:4]
+        if not year.isdigit():
+            raise ValueError("Issue has no usable cover date.")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)